sklearn/neural_network/extreme_learning_machines.py

```diff
@@ -0,0 +1,629 @@
+"""Extreme Learning Machines
+"""
+
+# Author: Issam H. Laradji <issam.laradji@gmail.com>
+# Licence: BSD 3 clause
+
+
+from abc import ABCMeta, abstractmethod
+
+import numpy as np
+from scipy import linalg
+
+from ..base import BaseEstimator, ClassifierMixin, RegressorMixin
+from .base import logistic, softmax, ACTIVATIONS
+from ..externals import six
+from ..preprocessing import LabelBinarizer
+from ..metrics import mean_squared_error
+from ..linear_model.ridge import ridge_regression
+from ..utils import gen_batches, check_random_state
+from ..utils import check_array, check_X_y, column_or_1d
+from ..utils.extmath import safe_sparse_dot
+from ..utils.class_weight import compute_sample_weight
+
+
+def _multiply_weights(X, sample_weight):
+    """Return W*X if sample_weight is not None."""
+    if sample_weight is None:
+        return X
+    else:
+        return X * sample_weight[:, np.newaxis]
+
+
+class BaseELM(six.with_metaclass(ABCMeta, BaseEstimator)):
+    """Base class for ELM classification and regression.
+
+    Warning: This class should not be used directly.
+    Use derived classes instead.
+    """
+    @abstractmethod
+    def __init__(self, n_hidden, activation, C, class_weight,
+                 weight_scale, batch_size, verbose, warm_start,
+                 random_state):
+        self.C = C
+        self.activation = activation
+        self.class_weight = class_weight
+        self.weight_scale = weight_scale
+        self.batch_size = batch_size
+        self.n_hidden = n_hidden
+        self.verbose = verbose
+        self.warm_start = warm_start
+        self.random_state = random_state
+
+    def _init_weights(self, n_features):
+        """Initialize the parameter weights."""
+        rng = check_random_state(self.random_state)
+
+        # Use the initialization method recommended by Glorot et al.
```

Comment:
This should be documented in the docstring / narrative. Also, is this good for ELM?